File: client/ios/App/App/VoiceCallPlugin.swift
import Foundation
import Capacitor

/// Capacitor plugin to manage background voice call state on iOS.
/// Background keep-alive is now handled by @anuradev/capacitor-background-mode plugin.
/// This plugin remains for API parity with Android's foreground service.
@objc(VoiceCallPlugin)
public class VoiceCallPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "VoiceCallPlugin"
    public let jsName = "VoiceCallService"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "start", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "stop", returnType: CAPPluginReturnPromise)
    ]

    @objc func start(_ call: CAPPluginCall) {
        call.resolve()
    }

    @objc func stop(_ call: CAPPluginCall) {
        call.resolve()
    }
}

File: client/android/app/src/main/java/net/mqvi/app/VoiceCallPlugin.java
package net.mqvi.app;

import android.content.Intent;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * Capacitor plugin to start/stop the VoiceCallService foreground service.
 * Called from JS when user joins/leaves a voice channel.
 */
@CapacitorPlugin(name = "VoiceCallService")
public class VoiceCallPlugin extends Plugin {

    @PluginMethod()
    public void start(PluginCall call) {
        Intent intent = new Intent(getContext(), VoiceCallService.class);
        getContext().startForegroundService(intent);
        call.resolve();
    }

    @PluginMethod()
    public void stop(PluginCall call) {
        Intent intent = new Intent(getContext(), VoiceCallService.class);
        getContext().stopService(intent);
        call.resolve();
    }
}
